pkg/deploy/deploy.go

```diff
@@ -488,4 +388,134 @@ func OperatorExists(cli client.Client, operatorPrefix string) (bool, error) {
 	return false, nil
 }
 
+func getResource(ctx context.Context, cli client.Client, obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	found := &unstructured.Unstructured{}
+	// Setting gvk is required to do Get request
+	found.SetGroupVersionKind(obj.GroupVersionKind())
+	err := cli.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, found)
+	return found, err
```

Comment:
It again ends up in `found` never nil.

Reply:
I think we would need to return `nil` when we get ObjectNotFound error

Reply:
Well, since we have error handling on the caller function, returning nil with any error should be ok (and actual object on non-error Get only)